dio/lib/src/dio_mixin.dart

```diff
@@ -456,7 +457,10 @@ abstract class DioMixin implements Dio {
 
         Future<InterceptorState> handleError() async {
           final errorHandler = ErrorInterceptorHandler();
-          interceptor(err.data, errorHandler);
+          final dioError = (err.data as DioError).copyWith(
+            stackTrace: stackTrace,
+          );
```

Comment:
Is this really needed? The correct stacktrace is seemingly already assigned in line 453/454

Reply:
It is only assigned if err is not of type InterceptorState already, maybe the proper solution is to always update err (line 452)?

Reply:
I've now updated the code to handle the DioError differently, please have a look at my last commit :)